crowbar_framework/app/models/node_object.rb

```diff
@@ -32,8 +32,13 @@ def self.find(search)
     if nodes.is_a?(Array) and nodes[2] != 0 and !nodes[0].nil?
       nodes[0].delete_if { |x| x.nil? }
       answer = nodes[0].map do |x|
-        NodeObject.new x
+        begin
+          NodeObject.new x
+        rescue Crowbar::Error::NotFound
+          nil
+        end
       end
+      answer.compact!
       answer.delete_if { |x| !x.has_chef_server_roles? }
```

Comment:
Hrm, actually, that line is supposed to deal with this already. But we don't use this `has_chef_server_roles?` method elsewhere while I guess we should.

That being said, I feel the exception is a better way to deal with this; I'd remove the `has_chef_server_roles?` bit instead.